packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,542 @@
+    }
+
+    final WindowingOwner owner = WidgetsBinding.instance.windowingOwner;
+    if (!owner.hasTopLevelWindows()) {
+      // TODO(mattkae): close the application if this is the last window
+      // via ServicesBinding.instance.exitApplication(AppExitType.cancelable);
+    }
+  }
+}
+
+/// A controller for a regular window.
+///
+/// A regular window is a traditional window that can be resized, minimized,
+/// maximized, and closed. Upon construction, the window is created for the
+/// platform with the provided properties.
+///
+/// This class does not interact with the widget tree. Instead, it is typically
+/// provided to the [RegularWindow] widget, who does the work of rendering the
+/// content inside of this window.
+///
+/// An example usage might look like:
+/// ```dart
+/// final RegularWindowController controller = RegularWindowController(
+///   contentSize: const WindowSizing(
+///     preferredSize: Size(800, 600),
+///     preferredConstraints: BoxConstraints(minWidth: 640, minHeight: 480),
+///   ),
+///   title: "Example Window",
+/// );
+/// runWidget(
+///   RegularWindow(
+///     controller: controller,
+///     child: MaterialApp(home: Container()),
+///   ),
+/// );
+/// ```
+///
+/// When provided to a [RegularWindow] widget, widgets inside of the [child]
+/// parameter will have access to the [RegularWindowController] via the
+/// [WindowControllerContext] widget.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+abstract class RegularWindowController extends BaseWindowController {
+  /// Creates a [RegularWindowController] with the provided properties.
+  ///
+  /// Upon construction, the window is created for the platform.
+  ///
+  /// [contentSize] sizing requests for the window. This might not behonored by the platform
+  /// [title] the title of the window
+  /// [state] the initial state of the window
+  /// [delegate] optional delegate for the controller controller.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  factory RegularWindowController({
+    required WindowSizing preferredContentSize,
+    String? title,
+    RegularWindowControllerDelegate? delegate,
+  }) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_windowingDisabledErrorMessage);
+    }
+
+    WidgetsFlutterBinding.ensureInitialized();
+    final WindowingOwner owner = WidgetsBinding.instance.windowingOwner;
+    final RegularWindowController controller = owner.createRegularWindowController(
+      preferredContentSize: preferredContentSize,
+      delegate: delegate ?? RegularWindowControllerDelegate(),
+    );
+    if (title != null) {
+      controller.setTitle(title);
+    }
+    return controller;
+  }
+
+  /// Creates an empty [RegularWindowController].
+  ///
+  /// TODO(mattkae): Replace @internal with @visibleForTesting when this API is non-experimental
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
```

Comment:
If you use a `//` for the TODO, it won't show up on the API reference :)

```suggestion
  /// Creates an empty [RegularWindowController].
  ///
  /// {@macro flutter.widgets.windowing.experimental}
  // TODO(mattkae): Replace @internal with @visibleForTesting when this API is non-experimental
```